lib/dao.js

```diff
@@ -2407,6 +2577,201 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+DataAccessObject.prototype.replaceAttributes = function(data, options, cb) {
+  var Model = this.constructor;
+  var id = getIdValue(this.constructor, this);
+  var idValue = idName(this.constructor)
+  data[idValue] = id;
```

Comment:
I think you need to set it inside `replaceById`, not only in `prototype.replaceAttributes`
at least if the following is a correct usage:

 ```
Customer.replaceById(123, { name: 'Miroslav' });
```

if ^^^ is not a valid usage (i.e. we require that data must contain the id field too)
then we must implement a validation and return an error when `data.id !== id`
(of course, use `idName` instead of `id` as the key)
thoughts?